ouroboros-consensus-shelley-test/src/Test/Consensus/Shelley/Generators.hs

```diff
@@ -55,21 +56,22 @@ import           Test.Shelley.Spec.Ledger.Serialisation.Generators ()
 
 -- | The upstream 'Arbitrary' instance for Shelley blocks does not generate
 -- coherent blocks, so neither does this.
-instance CanMockPreAlonzo era => Arbitrary (ShelleyBlock era) where
+instance CanMock era => Arbitrary (ShelleyBlock era) where
   arbitrary = mkShelleyBlock <$> arbitrary
 
 -- | This uses a different upstream generator to ensure the header and block
 -- body relate as expected.
-instance CanMockPreAlonzo era => Arbitrary (Coherent (ShelleyBlock era)) where
+instance CanMock era => Arbitrary (Coherent (ShelleyBlock era)) where
   arbitrary = Coherent . mkShelleyBlock <$> genCoherentBlock
 
-instance CanMockPreAlonzo era => Arbitrary (Header (ShelleyBlock era)) where
+instance CanMock era => Arbitrary (Header (ShelleyBlock era)) where
   arbitrary = getHeader <$> arbitrary
 
 instance SL.Mock c => Arbitrary (ShelleyHash c) where
   arbitrary = ShelleyHash <$> arbitrary
 
-instance CanMockPreAlonzo era => Arbitrary (GenTx (ShelleyBlock era)) where
+instance (CanMock era , Arbitrary (Core.Witnesses era))
```

Comment:
Is there anything preventing us from also moving `Arbitrary (Core.Witnesses era)` to `CanMock`? IIUC, it should hold for every Shelley-based era.

Reply:
Moved to `CanMock` and everything builds fine. Nice suggestion.